src/Parsers/ASTCheckQuery.h

```diff
@@ -26,29 +26,28 @@ struct ASTCheckQuery : public ASTQueryWithTableAndOutput
     QueryKind getQueryKind() const override { return QueryKind::Check; }
 
 protected:
-    void formatQueryImpl(const FormatSettings & settings, FormatState & state, FormatStateStacked frame) const override
+    void formatQueryImpl(FormattingBuffer out) const override
     {
-        std::string nl_or_nothing = settings.one_line ? "" : "\n";
-
-        std::string indent_str = settings.one_line ? "" : std::string(4 * frame.indent, ' ');
-        std::string nl_or_ws = settings.one_line ? " " : "\n";
-
-        settings.ostr << (settings.hilite ? hilite_keyword : "") << indent_str << "CHECK TABLE " << (settings.hilite ? hilite_none : "");
+        out.writeIndent();
+        out.writeKeyword("CHECK TABLE ");
 
         if (table)
         {
             if (database)
             {
-                settings.ostr << (settings.hilite ? hilite_keyword : "") << indent_str << backQuoteIfNeed(getDatabase()) << (settings.hilite ? hilite_none : "");
-                settings.ostr << ".";
+                out.writeIndent();
+                out.writeKeyword(backQuoteIfNeed(getDatabase()));
+                out.ostr << ".";
             }
-            settings.ostr << (settings.hilite ? hilite_keyword : "") << indent_str << backQuoteIfNeed(getTable()) << (settings.hilite ? hilite_none : "");
+            out.writeIndent();
+            out.writeKeyword(backQuoteIfNeed(getTable()));
```

Comment:
It seems a bit strange that database and table are formatted as keywords, but I see in master it was the same.

Reply:
Added this to the collection of formatting / hiliting issues #48140